checker/src/abstract_domains.rs

```diff
@@ -268,6 +269,34 @@ impl AbstractDomain {
         .into()
     }
 
+    /// Returns an element that is "self as target_type".
+    pub fn cast(&self, target_type: ExpressionType) -> Self {
+        if let Expression::CompileTimeConstant(v1) = &self.expression {
+            let result = v1.cast(target_type.clone());
+            if result != ConstantDomain::Bottom {
+                return result.into();
+            }
+        };
+        match &self.expression {
+            Expression::Bottom => self.clone(),
+            Expression::ConditionalExpression {
+                condition,
+                consequent,
+                alternate,
+            } => ConditionalExpression {
+                condition: condition.clone(),
+                consequent: box consequent.cast(target_type.clone()),
+                alternate: box alternate.cast(target_type),
+            }
+            .into(),
```

Comment:
Heh rustfmt does some interesting formatting here.

Reply:
Yeah, I really dislike this one, but I know better than trying to fight City Hall.